smack-extensions/src/main/java/org/jivesoftware/smackx/usertune/element/UserTuneElement.java

```diff
@@ -0,0 +1,258 @@
+        }
+        xml.rightAngleBracket();
+        xml.optAppend(getUserTuneXML());
+        return xml.closeElement(getElementName());
+    }
+
+    private CharSequence getUserTuneXML() {
+
+        XmlStringBuilder userTuneXML = new XmlStringBuilder();
+        userTuneXML.optElement("artist", artist);
+        userTuneXML.optElement("length", length);
+        userTuneXML.optElement("rating",rating);
+        userTuneXML.element("source",source);
+        userTuneXML.element("title",title);
+        userTuneXML.optElement("track",track);
+        userTuneXML.optElement("uri",uri);
+        return userTuneXML.toString();
+    }
+
+    private boolean isEmptyUserTune() {
+        if (this.artist == null && this.length == null && this.rating == null && this.source == null && this.title == null
+                && this.track == null && this.uri == null) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static boolean hasUserTuneElement(Message message) {
+        return message.hasExtension(ELEMENT, NAMESPACE);
+    }
+
+    public static UserTuneElement fromMessage(Message message) {
+        return message.getExtension(ELEMENT, NAMESPACE);
+    }
+
+    public UserTuneElement getUserTune() {
+        return this;
+    }
+
+    /**
+     * Returns a new instance of {@link Builder}.
+     * @return Builder
+     */
+    public static Builder getBuilder() {
+        return new Builder();
+    }
+
+    /**
+     * This class defines a Builder class for {@link UserTuneElement}.
+     * <br>
+     * {@link UserTuneElement} instance can be obtained using the {@link #build()} method as follows.
+     * <br>
+     * UserTuneElement.Builder builder = new UserTuneElement.Builder();
+     * builder.setSource("Yessongs"); builder.setTitle("Heart of the Sunrise");
+     * UserTuneElement userTuneElement = builder.build();
+     * <br>
+     * Values such as title, source, artist, length, source, track and uri can be set using their respective setters through {@link Builder}
+     * <br>
+     * Use {@link #buildEmptyUserTune()} only while disabling UserTune publish through {@link UserTuneManager#clearUserTune()}
+     */
+    public static final class Builder {
+        private String artist;
+        private Integer length;
+        private Integer rating;
+        private String source;
+        private String title;
+        private Integer track;
+        private URI uri;
+
+        private Builder() {
+        }
+
+        /**
+         * Artist is an optional element in UserTuneElement.
+         * @param artist
+         * @return Builder
+         */
+        public Builder setArtist(String artist) {
+            this.artist = artist;
+            return this;
+        }
+
+        /**
+         * Length is an optional element in UserTuneElement.
+         * @param length
+         * @return Builder
+         */
+        public Builder setLength(int length) {
+            if (length > 0) {
```

Comment:
This is not sufficient: 'length' is specified as xs:unsignedShort, you need to enforce that, which probably means that you want to add an according method to Smack's` `NumberUtil`.